modules/packages/freebsd_ports

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh -e
+
+command=$1
+
+while read -r line; do
+  export INPUT_$line
+done
+
+list_updates() {
+  pkg version -v -l "<" | sed -Ee $'s/([^ ]+)-[^-]+ .* (.*)\\)/Name=\\1\\\nVersion=\\2\\\nArchitecture=freebsd:10:x86:64/'
```

Comment:
This is the wrong way to do it. Correct way is via pkg query to extract relevant data.

```
$ pkg query '%n %v %q %R'
perl5 5.18.4_17 freebsd:10:x86:64 DragonNorth
$ pkg shell 
sqlite> select * from packages where name = 'perl5';
17|lang/perl5.18|perl5|5.18.4_17|Practical Extraction and Report Language|Perl is a language that combines some of the features of C, sed, awk and shell.  See the manual page for more hype.  There are also many books published by O'Reilly & Assoc.  See pod/perlbook.pod for more
information.

WWW: http://www.perl.org/|||FreeBSD:10:amd64|perl@FreeBSD.org|http://www.perl.org/|/usr/local|51917202|1|0|124|1445255055|2$0$jpuj587kyjfez7bcceod8r8m49wrye6gbg3j7igpnczzytfndify||
```

It is vitally important to NOT fake the architecture because pkg is multiplatform - not just FreeBSD. As I mentioned on IRC, a lot of the trickery here is going to be pkg query or pkg shell. They are often dramatically faster (2-4x) and will only present relevant information. Note that architecture must be handled *case insensitive* if you are mixing query and shell (shell -c "SQL Command"). The architecture needs to be parsed out into distinct fields as well. Breaking it down:

freebsd - OS
10 - major; major-minor is no longer present in pkg
x86 - architecture may be x86, arm, ia64, ppc, etc.
64 - endianness